woocommerce/woocommerce
# [NOX]: Add StatusBadge Component

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes https://github.com/woocommerce/woocommerce/issues/52555.

* Adds component needed by the new settings pages, `StatusBadge`.
* Adds logic to render the appropriate status badge.
* Note that this PR doesn't include tooltips, and incentive based badges, which will require a slightly different logic as we'll need to pass a message in to the component in this case.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

Pre-requisite:
- Have woocommerce installed locally and synced within WooPayments. See Pe4Hk0-3f-p2 for details. 
- Enable the feature flag to see reactified settings page. pe4Hk0-1ly-p2.

1. Locally, check out to `add/status-indicator-component`
2. Run `pnpm i && pnpm build`
3. Open the settings page http://localhost:8082/wp-admin/admin.php?page=wc-settings&tab=checkout
4. Verify that components look according to the design. (note: for this PR - just focus on the Status Badge) - see #52555  for link to design p2.

<img width="2191" alt="Screenshot 2024-11-04 at 20 52 40" src="https://github.com/user-attachments/assets/17ef4ad8-6eab-47f1-83e4-7838bce26262">
